### ISSUE TITLE
Add appropriate handling with custom runtime error in case of absence of a topic

### ISSUE THREAD

**Issue by oleksandr-balyshyn** [akka/alpakka-kafka#1385]
```
java.lang.NullPointerException: null
        at akka.kafka.internal.KafkaConsumerActor.$anonfun$handleMetadataRequest$3(KafkaConsumerActor.scala:559)
 ```
https://github.com/akka/alpakka-kafka/blob/eb5246d3e03992ce60a1f33fefe75dadc58bc866/core/src/main/scala/akka/kafka/internal/KafkaConsumerActor.scala#L668

partitionsFor.asScala can throw java.lang.NullPointerException, due to missing result in a Map.
partitionsFor - returns `null`
```
 @Override
    public List<PartitionInfo> partitionsFor(String topic) {
        acquireAndEnsureOpen();
        try {
            Cluster cluster = this.metadata.fetch();
            List<PartitionInfo> parts = cluster.partitionsForTopic(topic);
            if (!parts.isEmpty())
                return parts;

            Map<String, List<PartitionInfo>> topicMetadata = fetcher.getTopicMetadata(
                    new MetadataRequest.Builder(Collections.singletonList(topic), true), requestTimeoutMs);
            return topicMetadata.get(topic);
        } finally {
            release();
        }
    }
```
    
According to Fetcher implementation:
   

https://github.com/apache/kafka/blob/5936a249bda171851cfa669da4276d23230eab6e/clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java#L401-L404

**Comment by ennru** [comment]
Thank you for pointing this out! Would you be in a position to suggest a fix in a PR?

**Comment by limpid-kzonix** [comment]
@ennru I will prepare PR soon, just have to do some research on the current codebase (how to cover this change with tests)

**Comment by limpid-kzonix** [comment]
@ennru should be fixed with version 2.8.0 of kafka-clients

**Comment by ennru** [comment]
Note that Alpakka Kafka 2.1.0 is on kafka-clients 2.7.0.
Upgrade pending in https://github.com/akka/alpakka-kafka/pull/1364

But users are free to pull in a later version.
